pandas/tests/util/test_util.py

```diff
@@ -76,3 +76,16 @@ def test_rng_context():
         with tm.RNGContext(1):
             assert np.random.randn() == expected1
         assert np.random.randn() == expected0
+
+
+def test_external_error_raised():
+    # Raising an external error message
+    with tm.external_error_raised(TypeError):
+        pytest.approx()
+
+    # Note: Should be the only place in the codebase with
+    # "pytest.raises(Exception, match=None)"
+    # (Except pandas._testing.external_error_raised)
+    with pytest.raises(TypeError, match=None):
+        with tm.external_error_raised(ValueError):
+            pytest.approx()
```

Comment:
Can you explain what the purpose of this test is?

Reply:
To validate that the argument passed in, which is the exception to catch, is the __only__ exception it will try to catch.

to validate that this, won't pass.

```python
with tm.external_error_raised(ValueError):
    raise NotImplementedError("foo")
```

Can remove this test case, if it's more confusing than helping.

Reply:
I would drop the test.  The one you have above is sufficient.

Reply:
I would drop the test.  The one you have above is sufficient.